### PR TITLE
Move DefaultRegistry to Project Folder

### PR DESCRIPTION
**Pull request by brett-groshong**
### Purpose of this PR
- Put all Registry classes and interfaces in `UnityEditor.ShaderGraph.GraphDelta` namespace.
- Move `DefaultRegistry` into the `GraphUI` folder because it is the default for the application, not the library.
- Replace `com.unity.shadergraph` namespace with `UnityEditor.ShaderGraph` namespace.

---
### Testing status
These are largely namespace and code location changes. They are covered by the existing unit tests.

---
### Comments to reviewers
These changes are to facilitate changing when the `DefaultRegistry` is initialized.

### CODE REVIEW

File: com.unity.shadergraph/Tests/Generation/GenerationTests.cs [hunk 1]
```diff
@@ -48,20 +45,13 @@ private static Texture2D DrawToTex(Shader shader)
...
             Graphics.Blit(null, rt, new Material(shader));
-            Texture2D output = new Texture2D(4, 4, TextureFormat.ARGB32, false);
+            Texture2D output = new(4, 4, TextureFormat.ARGB32, false);
             output.ReadPixels(new Rect(0, 0, 4, 4), 0, 0);
             RenderTexture.active = prevActive;
             rt.Release();
             return output;
         }
 
-        static object[] testAsIsSource = new object[]
```
Comment (elizabeth-legros) [inline com.unity.shadergraph/Tests/Generation/GenerationTests.cs:58]: Why was this removed?

Comment (elizabeth-legros) [reply com.unity.shadergraph/Tests/Generation/GenerationTests.cs:58]: The test case source still refers to it

**Review by elizabeth-legros — CHANGES_REQUESTED** [review]
Fine other than removing test cases

File: com.unity.shadergraph/Tests/Generation/GenerationTests.cs (hunk 1 above)
Comment (elizabeth-legros) [reply com.unity.shadergraph/Tests/Generation/GenerationTests.cs:58]: If the intent was removing tests, that whole test needs to be removed alongside

Comment (brett-groshong) [reply com.unity.shadergraph/Tests/Generation/GenerationTests.cs:58]: I missed the reference in to this var in the attribute. Thanks for pointing it out.

Comment (brett-groshong) [reply com.unity.shadergraph/Tests/Generation/GenerationTests.cs]: Updated.

**Review by elizabeth-legros — APPROVED** [review]

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Artistic/Adjustment/ContrastNode.cs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 using System.Collections.Generic;
-using static UnityEditor.ShaderGraph.Registry.Types.GraphType;
+using Usage = UnityEditor.ShaderGraph.GraphDelta.GraphType.Usage;
 
-namespace com.unity.shadergraph.defs
+namespace UnityEditor.ShaderGraph.Defs
```
Comment (esmelusina) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Artistic/Adjustment/ContrastNode.cs:4]: Is there a case to have definitions live in a separate namespace?

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/Generation/Interpreter.cs [hunk 3]
```diff
@@ -290,13 +289,13 @@ private static bool FunctionsAreEqual(ShaderFunction a, ShaderFunction b)
...
             {
-                case 4: return $"{name}.xyz";
-                case 2: return $"float3({name}.x, {name}.y, 0)";
-                case 1: return $"float3({name}, {name}, {name})";
-                default: return name;
-            }
+                4 => $"{name}.xyz",
+                2 => $"float3({name}.x, {name}.y, 0)",
+                1 => $"float3({name}, {name}, {name})",
+                _ => name,
```
Comment (esmelusina) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/Generation/Interpreter.cs:297]: This syntax exists?

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/GraphDelta/GraphHandler.cs [hunk 4]
```diff
@@ -31,23 +29,42 @@ public string ToSerializedFormat()
...
         }
 
-        internal NodeHandler AddNode<T>(string name, Registry.Registry registry) where T : Registry.Defs.INodeDefinitionBuilder => graphDelta.AddNode<T>(name, registry);
+        internal NodeHandler AddNode<T>(string name, Registry registry) where T : INodeDefinitionBuilder =>
+            graphDelta.AddNode<T>(name, registry);
 
-        public NodeHandler AddNode(RegistryKey key, string name, Registry.Registry registry) => graphDelta.AddNode(key, name, registry);
-        public NodeHandler AddContextNode(RegistryKey key, Registry.Registry registry) => graphDelta.AddContextNode(key, registry);
-        public bool ReconcretizeNode(string name, Registry.Registry registry) => graphDelta.ReconcretizeNode(name, registry);
+        public NodeHandler AddNode(RegistryKey key, string name, Registry registry) =>
```
Comment (esmelusina) [inline com.unity.shadergraph/Editor/GraphDeltaRegistry/GraphDelta/GraphHandler.cs:35]: Style wise, for multi-line inline functions; I'm wondering if it's more clear if the => appears on the following line.

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Artistic/Adjustment/ContrastNode.cs (hunk 2 above)
Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Artistic/Adjustment/ContrastNode.cs:4]: Yes. When we spoke about the `FunctionDescriptor` concept, it seemed appropriate to separate it from the rest of the API. If it becomes a core ShaderGraph concept, we might want to move it into a shared namespace.

Secondarily, `Defs` namespace is problematic because it contains things like `Vector` and `Matrix`.

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/Generation/Interpreter.cs (hunk 3 above)
Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphDeltaRegistry/Generation/Interpreter.cs:297]: Yep! It's new to me too. However, this is the MS suggested syntax for switch statements that have a break for every case.

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/GraphDelta/GraphHandler.cs (hunk 4 above)
Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphDeltaRegistry/GraphDelta/GraphHandler.cs:35]: Stylistically, I have been updating inline functions to be this way because I feel it makes them read more like functions or methods. Also, one line in-lines can be pretty long.

File: com.unity.shadergraph/Editor/GraphDeltaRegistry/Generation/Interpreter.cs (hunk 3 above)
Comment (brett-groshong) [reply com.unity.shadergraph/Editor/GraphDeltaRegistry/Generation/Interpreter.cs:297]: Honestly, it surprises me that anonymous function creation is this cheap.
